src/main/kotlin/io/libp2p/core/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -0,0 +1,207 @@
+package io.libp2p.core.security.noise
+
+import com.google.protobuf.ByteString
+import com.southernstorm.noise.protocol.CipherState
+import com.southernstorm.noise.protocol.CipherStatePair
+import com.southernstorm.noise.protocol.DHState
+import com.southernstorm.noise.protocol.HandshakeState
+import io.libp2p.core.P2PAbstractChannel
+import io.libp2p.core.PeerId
+import io.libp2p.core.crypto.PrivKey
+import io.libp2p.core.crypto.unmarshalPublicKey
+import io.libp2p.core.multistream.Mode
+import io.libp2p.core.multistream.ProtocolMatcher
+import io.libp2p.core.security.SecureChannel
+import io.libp2p.etc.SECURE_SESSION
+import io.libp2p.etc.events.SecureChannelFailed
+import io.libp2p.etc.events.SecureChannelInitialized
+import io.netty.buffer.ByteBuf
+import io.netty.channel.ChannelHandlerContext
+import io.netty.channel.ChannelInboundHandlerAdapter
+import io.netty.channel.SimpleChannelInboundHandler
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.config.Configurator
+import spipe.pb.Spipe
+import java.util.concurrent.CompletableFuture
+
+class NoiseXXSecureChannel(val localKey: PrivKey, val localDHState: DHState, val remoteDHState: DHState, val role: Int) :
```

Comment:
We are expecting the only `PrivateKey` parameter passed for `SecureChannel`  according to the present API design (check out this `Host` construction sample: https://github.com/libp2p/jvm-libp2p/blob/7585e3e1560da26912f29cc142174c698ed6c6f3/src/test/kotlin/io/libp2p/core/HostTest.kt#L31)

- The `role` here can be taken from `P2PAbstractChannel.isInitiator`
- `remoteDHState` here is never used
- Couldn't the `localDHState` be derived from `Curve25519PrivateKey` ? If yes, then the only thing left to be passed here is the `Curve25519PrivateKey` instance

If the Noise requires something more than just a private key then we need to reconsider design

Reply:
Good points about the parameters that can be passed in, with required data generated from the keys. Grabbed the role from P2PAbstractChannel. Done.